### PR TITLE
[#52] Add docs

### LINKED ISSUE #864: Add installation section to documentation
We have to add next information into "installation" section:

1. Prerequirements
1.1. External Docker Registry - required for model storing
1.2. Monitoring (prometheus) chart - optional for components metrics
1.3. Performance metrics (statsd to prometheus) - optional for model metrics storing
1.4. Authorization - optional for WEB authorization
2. Step-by-step guide for available platforms
3. Detailed description of HELM charts (legion and legion infra), their values and examples of configuration

### LINKED ISSUE #869: Add network connectivity requirements for HELM chart
We have to describe what components have to have connection with which service (source namespaces, pods and target namespaces, pods and ports).

First assumptions are:
1. Kubernetes Ingress pod to EDGE/EDI pods port 80
2. Prometheus metrics collector to EDGE/EDI/Operator pods port XXX
3. EDGE pods to model pods

### LINKED ISSUE #877: Add detailed documentation about feedback gathering
As a user of Legion platform I want to have information how I can use feedback mechanism and how it works under the hood.
Also examples should be presented.

### PR DESCRIPTION
**Pull request by kirillmakhonin**
This updates #52 
This closes #864
This closes #869
This closes #877

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -63,30 +63,43 @@ install-robot:
 		python setup.py sdist && \
     	python setup.py bdist_wheel
 
-## docker-pipeline-agent: Build pipeline agent docker image
-docker-pipeline-agent:
-	docker build -t legion/python-pipeline:latest -f containers/pipeline/Dockerfile .
+## build-all-docker-images: build all docker images
```
Comment (mcdoker18) [inline Makefile]: Please, revert all changes from Makefile except doc building

File: containers/docs-builder/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+FROM python:3.5-alpine3.9
```
Comment (mcdoker18) [inline containers/docs-builder/Dockerfile]: > python:3.5
> 2019

File: docs/source/conf.py [hunk 3]
```diff
@@ -20,7 +20,7 @@
 import os
 import sys
 
-from legion.sdk.version import __version__ as legion_version
+#from legion.sdk.version import __version__ as legion_version
```
Comment (mcdoker18) [inline docs/source/conf.py:23]: Remove, please

File: docs/source/gs_deploying_model.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+```
+
+### Using `legionctl`
+
+```bash
+legionctl vcs create vcs-examlpe \
+          --type git \
+          --uri git@github.com:legion-platform/legion.git \
+          --default-reference master \
+          --credential "bG9sa2VrbG9sa2VrCg=="
```
Comment (mcdoker18) [inline docs/source/gs_deploying_model.md:44]: :)
```bash
➜  ✗ echo bG9sa2VrbG9sa2VrCg== | base64 -d
lolkeklolkek

```

File: docs/source/gs_deploying_model.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+```
+
+### Using Plain HTTP
+
+```bash
+curl -X POST "https://<edi-url>/api/v1/vcs" \
+     -H "accept: application/json" \
+     -H "Content-Type: application/json" \
+     -d '{ "name": "vcs-examlpe", "spec": { "credential": "bG9sa2VrbG9sa2VrCg==", "defaultReference": "master", "type": "git", "uri": "git@github.com:legion-platform/legion.git" }}'
```
Comment (mcdoker18) [inline docs/source/gs_deploying_model.md:53]: the same

File: docs/source/gs_deploying_model.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+### Using `kubectl`
+
+```bash
+kubectl create -f - <<EOF
+apiVersion: legion.legion-platform.org/v1alpha1
+kind: VCSCredential
+metadata:
+  name: vcs-examlpe
+spec:
+  credential: bG9sa2VrbG9sa2VrCg==
```
Comment (mcdoker18) [inline docs/source/gs_deploying_model.md:30]: the same

**Review by mcdoker18 — CHANGES_REQUESTED** [review]

File: docs/source/gen_used_technologies.rst [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+Technologies Used
+=====
+
+- **Python Language** - for some of build toolchains, model packagers and deploying integrations.
+
+- **Go Language** - for platform orchestrating applications (in tandem with `dep <https://golang.github.io/dep/>`_) for dependency management).
+
+- **Kubernetes** – as variants of runtime platform.
+
+- **HELM** - for disturbing Legion as Kubernetes application.
```
Comment (maxkochubey) [inline docs/source/gen_used_technologies.rst]: "Distributing"? :)

**Review by mcdoker18 — DISMISSED** [review]

File: docs/source/gen_used_technologies.rst (hunk 7 above)
Comment (mcdoker18) [reply docs/source/gen_used_technologies.rst]: Fixed

**Review by mcdoker18 — DISMISSED** [review]

File: README.MD [hunk 8]
```diff
@@ -1,43 +1,37 @@
...
+
+* **Open** - Legion is an open source project with its friendly community on [GitHub](https://github.com/legion-platform/legion).
+* **Comprehensible** - you can install Legion with minimum effort and track performance of ML models internals and compare their predictions.
+* **Flexible** - well-versed user can manage and control ML model building and training, at the same time, newcomers will find it easy to use.
+* **Scalable** - you can use existing and new models with the added architecture and feature sets.
 
-Legion provides: 
+The platform was designed in such a way that it contains the capabilities needed for model development purposes using supported [toolchains](docs/source/toolchains.md) (Python) as well as for the platform deployment (onto Kubernetes Cluster).
+
+It also includes the components that are deployed on the cluster (**Graphana**, **Prometheus**) and the ones that are required for that deployment (**Ansible**, **Helm**) or building Legion's components from source code (**Jenkins files**).
```
Comment (maxkochubey) [inline README.MD]: - Grafana
- Ansible was replaced by Terraform

**Review by maxkochubey — CHANGES_REQUESTED** [review]

File: docs/source/dev_project_structure.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+* `docs` - documentation that describes Legion platform, architecture, usage and etc.
+* `examples` - examples of machine learning models that could be trained and deployed in Legion, examples base on public available models (such as sklearn digit classifier, MovieLens model, Logistic Regression classifier on Census Income Data) and some syntetic models. [Details](examples/README.md)
+* `helms` - Legion Helm packages (distribution packages for Kubernetes).
+* `legion` - source code of Legion python packages.
+* `scripts` - utilitary scripts for CI/CD process.
+
+## Infra-specific repositories
+For deploying purposes there are platform-specific repositories that contains platform-specific deploying logic.
+
+**Legion AWS** locates in GitHub Repository [legion-platform/legion-aws](https://github.com/legion-platform/legion-aws) and contains:
```
Comment (maxkochubey) [inline docs/source/dev_project_structure.md]: Does this information is relevant (ansible-related below too)?

File: docs/source/gen_security.rst [hunk 10]
```diff
@@ -0,0 +1,49 @@
+===================
+Security approaches
+===================
+
+Legion platform does not manages users, tokens or other credentials by itself. Instead, it relies on OAuth2 IP (identity provider), that can be deployed anyware.
```
Comment (maxkochubey) [inline docs/source/gen_security.rst]: Typo: anywhere

**Review by maxkochubey — CHANGES_REQUESTED** [review]

File: README.MD (hunk 8 above)
Comment (mcdoker18) [reply README.MD]: Fixed

File: docs/source/gen_security.rst (hunk 10 above)
Comment (mcdoker18) [reply docs/source/gen_security.rst]: Fixed

File: docs/source/dev_project_structure.md (hunk 9 above)
Comment (mcdoker18) [reply docs/source/dev_project_structure.md]: Fixed

**Review by mcdoker18 — APPROVED** [review]

**Review by maxkochubey — APPROVED** [review]

**Pull request merged**